pinecone/core/grpc/index_grpc.py

```diff
@@ -155,42 +153,128 @@ def wrapped():
+        timeout = kwargs.pop('timeout', None)
+        response = self._wrap_grpc_call(self.stub.Upsert, request, timeout=timeout)
+        json_response = json_format.MessageToDict(response)
+        return parse_upsert_response(json_response)
 
-    def delete(self,
-               request: 'vector_service_pb2.DeleteRequest',
-               timeout: int = None,
-               metadata: Dict[str, str] = None):
-        return self._wrap_grpc_call(self.stub.Delete, request, timeout=timeout, metadata=metadata)
+    @sentry
+    @validate_and_convert_errors
+    def delete(self, *args, **kwargs):
+        request = DeleteRequest(*args, **kwargs)
+        timeout = kwargs.pop('timeout', None)
+        response = self._wrap_grpc_call(self.stub.Delete, request, timeout=timeout)
+        return response
 
-    def fetch(self,
-              request: 'vector_service_pb2.FetchRequest',
-              timeout: int = None,
-              metadata: Dict[str, str] = None):
-        return self._wrap_grpc_call(self.stub.Fetch, request, timeout=timeout, metadata=metadata)
+    @sentry
+    @validate_and_convert_errors
+    def fetch(self, *args, **kwargs):
+        timeout = kwargs.pop('timeout', None)
+        request = FetchRequest(*args, **kwargs)
+        response = self._wrap_grpc_call(self.stub.Fetch, request, timeout=timeout)
+        json_response = json_format.MessageToDict(response)
+        return parse_fetch_response(json_response)
 
-    def query(self,
-              request: 'vector_service_pb2.QueryRequest',
-              timeout: int = None,
-              metadata: Dict[str, str] = None):
-        return self._wrap_grpc_call(self.stub.Query, request, timeout=timeout, metadata=metadata)
+    @sentry
+    @validate_and_convert_errors
+    def query(self, queries, **kwargs):
+        timeout = kwargs.pop('timeout', None)
+
+        def _query_transform(item):
+            if isinstance(item, GRPCQueryVector):
+                return item
+            if isinstance(item, tuple):
+                values, filter = fix_tuple_length(item, 2)
+                filter = dict_to_proto_struct(filter)
+                return GRPCQueryVector(values=values, filter=filter)
+            if isinstance(item, Iterable):
+                return GRPCQueryVector(values=item)
+            raise ValueError(f"Invalid query vector value passed: cannot interpret type {type(item)}")
+
+        _QUERY_ARGS = ['namespace', 'top_k', 'filter', 'include_values', 'include_metadata']
+        if 'filter' in kwargs:
+            kwargs['filter'] = dict_to_proto_struct(kwargs['filter'])
+        request = QueryRequest(queries=list(map(_query_transform, queries)),
+                               **{k: v for k, v in kwargs.items() if k in _QUERY_ARGS})
+
+        response = self._wrap_grpc_call(self.stub.Query, request, timeout=timeout)
+        json_response = json_format.MessageToDict(response)
```

Comment:
This may be hurting performance a lot to convert from protobuf to json just to then convert to the response model class. Might defeat the purpose of grpc/protobuf somewhat. Something to keep in mind - anyways we'll see when we performance test the grpc client vs the json client.

Reply:
Yeah I have been wondering what the impact for this would be. But to get the responses to be human-readable we need to spend some effort in conversions. I will do a quick comparison to see if things are any different